src/routes/transactions/mappers/common/human-description.mapper.ts

```diff
@@ -0,0 +1,108 @@
+import { Inject, Injectable } from '@nestjs/common';
+import {
+  decodeFunctionData,
+  getFunctionSelector,
+  parseAbi,
+  formatUnits,
+  isHex,
+} from 'viem';
+import { ITokenRepository } from '../../../../domain/tokens/token.repository.interface';
+import { TokenRepository } from '../../../../domain/tokens/token.repository';
+import { Token } from '../../../../domain/tokens/entities/token.entity';
+import { MAX_UINT256 } from '../../../../routes/transactions/constants';
+import {
+  ILoggingService,
+  LoggingService,
+} from '../../../../logging/logging.interface';
+import { shortenAddress } from '../../../common/utils/utils';
+import { SafeAppInfo } from '../../../transactions/entities/safe-app-info.entity';
+import { IHumanDescriptionApi } from '../../../../domain/interfaces/human-description-api.interface';
+import {
+  HumanReadableFragment,
+  ValueType,
+} from '../../../../datasources/human-description-api/entities/human-description.entity';
+
+@Injectable()
+export class HumanDescriptionMapper {
+  constructor(
+    @Inject(ITokenRepository) private readonly tokenRepository: TokenRepository,
+    @Inject(LoggingService) private readonly loggingService: ILoggingService,
+    @Inject(IHumanDescriptionApi)
+    private readonly humanDescriptionApiService: IHumanDescriptionApi,
+  ) {}
+
+  async mapHumanDescription(
+    to: string | undefined,
+    data: string | null,
+    chainId: string,
+    safeAppInfo: SafeAppInfo | null,
+  ): Promise<string | undefined> {
+    if (!data || !isHex(data) || !to) return;
+
+    const parsedMessages = Object.entries(
+      this.humanDescriptionApiService.getParsedMessages(),
+    );
+
+    for (const [callSignature, template] of parsedMessages) {
+      const sigHash = getFunctionSelector(callSignature);
```

Comment:
This is called for every transaction that has data which can impact the performance as `getFunctionSelector` relies on hashing and string manipulation.

For readability I understand why we have the functions as a key in JSON format. But we should do some lazy evaluation for each one once we know the hash for each one.

Reply:
I've moved this into `HumanDescriptionRepository` as discussed so we don't have to compute on every call anymore.